store/siren-action-behavior.html

```diff
@@ -122,17 +133,17 @@
 			} else {
 				fields = this.getSirenFields(action);
 			}
-			if (action.method !== 'GET' && action.method !== 'HEAD') {
-				body = fields;
-			}
 
-			if (body && action.type.indexOf('json') !== -1) {
+			if (action.type.indexOf('json') !== -1) {
 				var json = {};
-				body.keys().forEach(function(key) {
-					json[key] = action.getFieldByName(key).value;
-				});
+				for (var i = 0; i < fields.length; i++) {
+					var field = fields[i];
+					json[field.name] = field.value;
+				}
 				headers.set('Content-Type', action.type);
 				body = JSON.stringify(json);
+			} else if (action.method !== 'GET' && action.method !== 'HEAD') {
+				body = new FormData(this._sequenceFields(fields));
```

Comment:
I don't think you can do this with `FormData`.

https://developer.mozilla.org/en-US/docs/Web/API/FormData/FormData

Reply:
right, got this mixed up with the url search params

Reply:
`_createFormData`?

Reply:
What happens if the method is 'GET' or 'HEAD'?